### PR TITLE
Remove Ice.Override.Timeout in C#

### PR DESCRIPTION
**Pull request by bernardnormier**
The C# counterpart of #2227.

### CODE REVIEW

File: csharp/src/Ice/Internal/ConnectionFactory.cs [hunk 1]
```diff
@@ -110,15 +110,10 @@ public void waitUntilFinished()
         }
     }
 
-    public void create(EndpointI[] endpts, bool hasMore, Ice.EndpointSelectionType selType,
+    public void create(List<EndpointI> endpoints, bool hasMore, Ice.EndpointSelectionType selType,
```
Comment (pepone) [inline csharp/src/Ice/Internal/ConnectionFactory.cs:113]: Why did you change the type from array to list? Should we use IList instead?

**Review by pepone — APPROVED** [review]

File: csharp/src/Ice/Internal/ConnectionFactory.cs (hunk 1 above)
Comment (bernardnormier) [reply csharp/src/Ice/Internal/ConnectionFactory.cs:113]: In the existing code, we "applied overrides" on the array which returned a new `List<EndpointI>`, and then passed this endpoint list to various APIs that expect an `List<EndpointI>`,  not an `IList<EndpointI>`. I didn't try to optimize this code.

**Review by externl — APPROVED** [review]
